webapp/channels/src/components/convert_gm_to_channel_modal/convert_gm_to_channel_modal.tsx

```diff
@@ -0,0 +1,218 @@
+import {UserProfile} from '@mattermost/types/users';
+import {displayUsername} from 'mattermost-redux/utils/user_utils';
+import {Team} from '@mattermost/types/teams';
+import {Client4} from 'mattermost-redux/client';
+import TeamSelector from 'components/convert_gm_to_channel_modal/team_selector/team_selector';
+import {trackEvent} from 'actions/telemetry_actions';
+import loadingIcon from 'images/spinner-48x48-blue.apng';
+import classNames from 'classnames';
+import NoCommonTeamsError from 'components/convert_gm_to_channel_modal/no_common_teams/no_common_teams';
+import AllMembersDeactivatedError from 'components/convert_gm_to_channel_modal/all_members_deactivated/all_members_deactivated';
+
+export type Props = {
+    onExited: () => void;
+    channel: Channel;
+    actions: Actions;
+    profilesInChannel: UserProfile[];
+    teammateNameDisplaySetting: string;
+    channelsCategoryId: string | undefined;
+}
+
+const ConvertGmToChannelModal = (props: Props) => {
+    const intl = useIntl();
+    const {formatMessage} = intl;
+
+    const handleCancel = useCallback(() => {
+        props.actions.closeModal(ModalIdentifiers.CONVERT_GM_TO_CHANNEL);
+    }, [props.actions.closeModal]);
+
+    const [channelName, setChannelName] = useState<string>('');
+    const handleChannelNameChange = useCallback((newName: string) => {
+        setChannelName(newName);
+    }, []);
+
+    const [channelURL, setChannelChannelURL] = useState<string>('');
+    const handleChannelURLChange = useCallback((newURL: string) => {
+        setChannelChannelURL(newURL);
+    }, []);
+
+    const channelMemberNames = props.profilesInChannel.map((user) => displayUsername(user, props.teammateNameDisplaySetting));
+
+    const [commonTeamsById, setCommonTeamsById] = useState<{[id: string]: Team}>({});
+    const [commonTeamsFetched, setCommonTeamsFetched] = useState<boolean>(false);
+    const [loadingAnimationTimeout, setLoadingAnimationTimeout] = useState<boolean>(false);
+    const [selectedTeamId, setSelectedTeamId] = useState<string>();
+    const [nameError, setNameError] = useState<boolean>(false);
+    const [conversionError, setConversionError] = useState<string>();
+
+    const handleTeamChange = useCallback((teamId: string) => {
+        setSelectedTeamId(teamId);
+    }, []);
+
+    useEffect(() => {
+        const work = async () => {
+            const teams = (await Client4.getGroupMessageMembersCommonTeams(props.channel.id)).data;
+            const teamsById: {[id: string]: Team} = {};
+            teams.forEach((team) => {
+                teamsById[team.id] = team;
+            });
+            setCommonTeamsById(teamsById);
+            setCommonTeamsFetched(true);
+
+            // if there is only common team, selected it.
+            if (teams.length === 1) {
+                setSelectedTeamId(teams[0].id);
+            }
+        };
+
+        work();
+        setTimeout(() => setLoadingAnimationTimeout(true), 1200);
```

Comment:
So lets say you open the modal, we show the loading indicator, then say, 150 ms later the data load from API is completed. So we remove the loading indicator and show the final UI. In such cases, the user briefly sees the loading indicator, aka flashing. Its a common UX practice to show loading state for a certain minimum time even if data load is finished to avoid this flashing. See example below.

https://github.com/mattermost/mattermost/assets/18575143/d9e5197a-3545-48ab-8901-2a1c0e420f0f

Since this isn't really critical part of the feature, in the interest of time, LMK whatever you want to do finally - keep this delay or remove it.

Reply:
Ah, I see. I'm not opposed to the idea then, although I feel like I'd also describe it as flashing when it's that quick.

I feel like I've been trained to be scared whenever I see `setTimeout` 😛